backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/search/impl/ElasticsearchSearchScopeModel.java

```diff
@@ -78,49 +78,51 @@ public EventContext getIndexesEventContext() {
 			IndexSchemaFieldNodeComponentRetrievalStrategy<T> componentRetrievalStrategy) {
 		ElasticsearchIndexModel indexModelForSelectedSchemaNode = null;
 		ElasticsearchIndexSchemaFieldNode<?> selectedSchemaNode = null;
-		T selectedComponent = null;
-		ElasticsearchConverterCompatibilityChecker converterChecker = null;
+		ElasticsearchScopedIndexFieldComponent<T> scopedIndexFieldComponent = new ElasticsearchScopedIndexFieldComponent<>();
 
 		for ( ElasticsearchIndexModel indexModel : indexModels ) {
 			ElasticsearchIndexSchemaFieldNode<?> schemaNode = indexModel.getFieldNode( absoluteFieldPath );
+			if ( schemaNode == null ) {
+				continue;
+			}
 
-			if ( schemaNode != null ) {
-				T component = componentRetrievalStrategy.extractComponent( schemaNode );
+			T component = componentRetrievalStrategy.extractComponent( schemaNode );
+			if ( selectedSchemaNode == null ) {
+				selectedSchemaNode = schemaNode;
+				indexModelForSelectedSchemaNode = indexModel;
+				scopedIndexFieldComponent.setComponent( component );
+				continue;
+			}
 
-				if ( selectedSchemaNode == null ) {
-					selectedSchemaNode = schemaNode;
-					indexModelForSelectedSchemaNode = indexModel;
-					selectedComponent = component;
-				}
-				else if ( !componentRetrievalStrategy.hasCompatibleCodec( selectedComponent, component ) ) {
-					throw componentRetrievalStrategy.createCompatibilityException(
-							absoluteFieldPath,
-							selectedComponent,
-							component,
-							EventContexts.fromIndexNames(
-									indexModelForSelectedSchemaNode.getHibernateSearchIndexName(),
-									indexModel.getHibernateSearchIndexName()
-							)
-					);
-				}
-				else if ( !componentRetrievalStrategy.hasCompatibleConverter( selectedComponent, component ) ) {
-					converterChecker = new ElasticsearchFailingConverterCompatibilityChecker<>(
-							absoluteFieldPath, selectedComponent, component, EventContexts.fromIndexNames(
-							indexModelForSelectedSchemaNode.getHibernateSearchIndexName(),
-							indexModel.getHibernateSearchIndexName()
-					), componentRetrievalStrategy );
-				}
+			if ( !componentRetrievalStrategy.hasCompatibleCodec( scopedIndexFieldComponent.getComponent(), component ) ) {
+				throw componentRetrievalStrategy.createCompatibilityException(
+						absoluteFieldPath,
+						scopedIndexFieldComponent.getComponent(),
+						component,
+						EventContexts.fromIndexNames(
+								indexModelForSelectedSchemaNode.getHibernateSearchIndexName(),
+								indexModel.getHibernateSearchIndexName()
+						)
+				);
+			}
+			ElasticsearchFailingCompatibilityChecker failingCompatibilityChecker = new ElasticsearchFailingCompatibilityChecker<>(
+					absoluteFieldPath, scopedIndexFieldComponent.getComponent(), component, EventContexts.fromIndexNames(
+					indexModelForSelectedSchemaNode.getHibernateSearchIndexName(),
+					indexModel.getHibernateSearchIndexName()
+			), componentRetrievalStrategy );
```

Comment:
This is indented quite strangely. Even automatic formatting in IDEA (which usually isn't great) looks better.